### ISSUE TITLE
UserDetails Missing Headings and Landmark Regions 

### ISSUE THREAD

**Issue by johnhbenetech** [benetech/MathShare#793]
This page has no main heading to communicate its purpose.  In addition, the teacher/student buttons have no heading preceding them, nor are the control groups headed on the teacher details form.  Finally, none of the content on this page (aside from the footer) is contained inside an ARIA landmark region. 
 
Suggested Solution 
 
Mark up the text, “Let's setup your account”, as a heading at level 1.  Then, add a level 2 heading before the teacher/student buttons with text such as, “Who are you?” or similar.  Also add a level 2 heading for both control group labels on the teacher details form (“What grade do you work with?” and “Which best describes your role?”). 
 
Finally, wrap the logo image inside a < header> element (or < div> with a role of “banner”), and everything between the < header> and < footer> inside a < main>.

Aha! Link: https://diagramlabs.aha.io/features/B-125

**Comment by jscholes** [comment]
@johnhbenetech that's not ideal.  @rupeshparab, can it be accessed via a specific URL route which will just override what's already been entered when submitted?  I don't need a control to invoke it from the UI, just a URL.

**Comment by rupeshparab** [comment]
@jscholes 
Currently, I have not added validation in the UI to prevent users who have submitted the data from revisiting the page.
You can access it using: https://mathshare-qa.diagramcenter.org/#/userDetails
Also, if you have already filled the details, the api call on submission might fail, but you will still see a success message as it will fail silently and allow user to continue to the dashboard

**Comment by jscholes** [comment]
@rupeshparab - thanks for that.  Some outstanding things to clean up:

* There is no page title for this route.
* The teacher/student buttons are not programmatically grouped.
    * There is a `<label>` element before them, but that pattern only applies to single form fields, not buttons or groups of them.
    * We can just use the same `<fieldset>` and `<legend>` pattern as used for the grade level controls.
* There is no accessible feedback when this page is loaded, nor when moving between the two steps.
* When a successful submission is made, the "thanks" alert and dashboard title end up interrupting each other.  Would it be possible to combine them, e.g., a single speech announcement of "Thanks for submitting your details. You are now on the problem sets dashboard" or something?

**Comment by rupeshparab** [comment]
I have fixed the 4 mentioned points, also this is the message on redirection after Submission of user details form:
'Thanks for sharing your details. You are now on "{pageTitle}" page'
pageTitle is being read from Page Title in DOM

**Comment by jscholes** [comment]
@rupeshparab:

1. When I activate the teacher button, I hear an alert, "Please fill your details".  An alert shouldn't be used here, because the message is only informational and isn't being delivered in exceptional circumstances.  Instead, the page should just announce the text of the first question ("What grade do you work with?") and focus the question's heading.
2. When I've completed and submitted the details, there is an alert fired informing me that I'm on the "All Problem Sets" page.  But it's interrupted by another alert which seems to be empty.
3. There also seems to be some unintended focus behaviour, because my screen reader (NVDA) is announcing the word "heading" a lot.  It's difficult to know why, until the blank alerts have been removed.
4. The text of the alert, "You are now on `<title>` page", needs the word "the" before the page title.

**Comment by rupeshparab** [comment]
I have pushed a fix for points 1 and 4
But I wasn't able to recreate the second issue of empty alert while using VoiceOver, will retry it with NVDA and let you know if I am still not able to encounter it.
For the 3rd issue, we have a screen reader only heading on the /app page, it is coming into focus once the success alert ends and is reading: 'heading Level 1 Dashboard' on Voiceover.

**Comment by rupeshparab** [comment]
I was able to recreate the heading announcement issue in alert, it was due to alert having only a message and not a heading, in case of VoiceOver the empty heading was being skipped.
All of the 4 issues have been resolved now

**Comment by jscholes** [comment]
@rupeshparab - thinking about this a bit more, I think we should remove the alert and just focus the `<h2>` with the text "Thanks for sharing your details".  And then only hide that message once the user has chosen to do something else (e.g. go to a problem set).  This would also mean not automatically setting focus to the "Dashboard" heading.

**Comment by rupeshparab** [comment]
@jscholes 
When you say "we should remove the alert and just focus the h2", does it mean that the alert popup will be removed and we will have a h2 element on dashboard with the text "Thanks for sharing your details" which will be focused?

**Comment by jscholes** [comment]
@rupeshparab yes.

**Comment by rupeshparab** [comment]
I have pushed a fix for this

**Comment by johnhbenetech** [comment]
@bpmcneilly please review changes in QA when you get a chance

**Comment by jscholes** [comment]
Fixed in https://github.com/benetech/MathShare/commit/1d87873b0d343171f7ac1ca09fda1c1d067b299d.
